optd-cost-model/src/cost_model.rs

```diff
@@ -20,28 +21,31 @@ use crate::{
 pub struct CostModelImpl<S: CostModelStorageLayer> {
     storage_manager: CostModelStorageManager<S>,
     default_catalog_source: CatalogSource,
+    _memo: Arc<dyn MemoExt>,
 }
```

Comment:
Not sure whether it's good to use `Arc` here.

Reply:
This looks fine, otherwise you'll need to add the memo type as part of the cost model impl's generic type parameters.